pkg/agent/datapath/natBridge.go

```diff
@@ -0,0 +1,515 @@
+		return err
+	}
+	if err := defaultFlow.Next(ofctrl.NewEmptyElem()); err != nil {
+		log.Errorf("Failed to install flow in CTState table %d: %s", NatBrCTStateTable, err)
+		return err
+	}
+
+	return nil
+}
+
+func (n *NatBridge) initSessionAffinityTable() error {
+	defaultFlow, err := n.sessionAffinityTable.NewFlow(ofctrl.FlowMatch{
+		Priority: DEFAULT_FLOW_MISS_PRIORITY,
+	})
+	if err != nil {
+		log.Errorf("Failed to new a flow in SessionAffinity table %d: %s", NatBrSessionAffinityTable, err)
+		return err
+	}
+	if err := defaultFlow.LoadField(ChooseBackendFlagReg, uint64(NeedChoose), ChooseBackendFlagRange); err != nil {
+		log.Errorf("Failed to add a load field action to flow in SessionAffinity table %d: %s", NatBrSessionAffinityTable, err)
+		return err
+	}
+	if err := defaultFlow.Next(ofctrl.NewEmptyElem()); err != nil {
+		log.Errorf("Failed to install flow in SessionAffinity table %d: %s", NatBrSessionAffinityTable, err)
+		return err
+	}
+	return nil
+}
+
+func (n *NatBridge) initServiceLBTable() error {
+	flow, err := n.serviceLBTable.NewFlow(ofctrl.FlowMatch{
+		Priority: HIGH_MATCH_FLOW_PRIORITY,
+		Regs: []*ofctrl.NXRegister{
+			{
+				RegID: constants.OVSReg0,
+				Data:  uint32(NoNeedChoose),
+				Range: openflow13.NewNXRange(16, 16),
+			},
+		},
+	})
+	if err != nil {
+		log.Errorf("Failed to new a flow in ServiceLB table %d: %s", NatBrServiceLBTable, err)
+		return err
+	}
+	if err = flow.Resubmit(nil, &NatBrDnatTable); err != nil {
+		log.Errorf("Failed to add resubmit action to flow in ServiceLB table %d: %s", NatBrServiceLBTable, err)
+		return err
+	}
+	if err = flow.Next(ofctrl.NewEmptyElem()); err != nil {
+		log.Errorf("Failed to install flow in ServiceLB table %d: %s", NatBrServiceLBTable, err)
+		return err
+	}
+	return nil
+}
+
+func (n *NatBridge) buildLearnActOfSessionAffinityLearnTable(ipProto uint8) (*ofctrl.LearnAction, error) {
+	ethTypeField := ofctrl.LearnField{Name: "nxm_of_eth_type", Start: 0}
+	ipSrcField := ofctrl.LearnField{Name: "nxm_of_ip_src", Start: 0}
+	ipDstField := ofctrl.LearnField{Name: "nxm_of_ip_dst", Start: 0}
+
+	ipProtoField := ofctrl.LearnField{Name: "nxm_of_ip_proto", Start: 0}
+	tcpDstField := ofctrl.LearnField{Name: "nxm_of_tcp_dst", Start: 0}
+	udpDstField := ofctrl.LearnField{Name: "nxm_of_udp_dst", Start: 0}
+
+	backendIPField := ofctrl.LearnField{Name: BackendIPReg, Start: 0}
+	backendPortField := ofctrl.LearnField{Name: BackendPortReg, Start: 0}
+	chooseBackendFlagField := ofctrl.LearnField{Name: ChooseBackendFlagReg, Start: uint16(ChooseBackendFlageStart)}
+
+	cookieID, err := getLearnCookieID()
+	if err != nil {
+		return nil, err
+	}
+	learnAct := ofctrl.NewLearnAction(NatBrSessionAffinityTable, MID_MATCH_FLOW_PRIORITY, 0, 300, 0, 0, cookieID)
+	learnAct.SetDeleteLearned()
+
+	if err := learnAct.AddLearnedMatch(&ethTypeField, 16, nil, uintToByteBigEndian(uint16(PROTOCOL_IP))); err != nil {
```

Comment:
nit: replace the magic number with a named constant

Reply:
done